Source/WebKit/NetworkProcess/NetworkLoadChecker.cpp

```diff
@@ -214,7 +214,7 @@ ResourceError NetworkLoadChecker::validateResponse(const ResourceRequest& reques
     if (response.containsInvalidHTTPHeaders())
         return badResponseHeadersError(request.url());
 
-    auto scope = makeScopeExit([&] {
+    auto scope = makeScopeExit([this, protectedThis = Ref { *this }, &response] {
```

Comment:
@rniwa don't we want to add `NOESCAPE` for the parameter to makeScopeExit instead?

Reply:
I see this pattern in a few places, so not sure what to do.

```
Source/WebKit/WebProcess/WebPage/WebFrame.cpp:    return makeScopeExit<Function<void()>>([protectedThis = Ref { *this }] {
Source/WebKit/UIProcess/WebProcessProxy.cpp:    auto scope = makeScopeExit([this, protectedThis = Ref { *this }]() {
Source/WebKit/WebProcess/WebPage/WebHistoryItemClient.cpp:    return makeScopeExit(CompletionHandler<void()> { [this, protectedThis = Ref { *this }] {
Source/WebKit/UIProcess/AuxiliaryProcessProxy.cpp:    auto scopeExit = WTF::makeScopeExit([protectedThis = Ref { *this }] {
```

Reply:
I don't think we can add `NOESCAPE` for `makeScopeExit` since the point of `makeScopeExit` to create an object which runs the function in the destructor, meaning that it stores the lambda.